src/AutoVizuA11y.tsx

```diff
@@ -239,92 +219,113 @@ const AutoVizuA11y = ({
-							}
-						},
-					);
+					generateDescriptions({
+						title,
+						data,
+						average: averageAux,
+						context,
+						apiKey: autoDescriptions!.apiKey,
+						model: autoDescriptions!.model,
+						temperature: autoDescriptions!.temperature,
+					}).then(function (result) {
+						chartDescriptions = result; // Output: [longerDescValue, smallerDescValue]
+						setDescs(result); // Output: [longerDescValue, smallerDescValue]
+						descriptionsKeyHandler({
+							chartRef,
+							setDescriptionContent,
+							type,
+							descs: chartDescriptions,
+							title,
+							autoDescriptions,
+						});
+
+						if (autoDescriptions && autoDescriptions.dynamicDescriptions === false) {
+							localStorage.setItem(storedLongerKey, chartDescriptions[0]);
+							localStorage.setItem(storedSmallerKey, chartDescriptions[1]);
+						}
+					});
 				}
 			});
 
-			//wipes the old tabindex present in the child components
-			levelChart(ref, true);
+			// sets the navigation onto the charts first
+			switchToChartLevel(chartRef, true);
 		}, 500);
+	}, [chartRef]);
 
-		//creates the ShortcutGuide
-		createShortcutGuide();
-	}, [ref]);
-
-	//sets the appropriate navigation keys in the ShortcutGuide
-	function handleNav(event: React.KeyboardEvent<HTMLDivElement>) {
-		levelNav(event, ref);
-	}
-
-	//sets the appropriate navigation keys and shortcuts in the charts and data
-	function handleKeyDown(event: React.KeyboardEvent<HTMLDivElement>) {
-		let alertDiv: HTMLElement | null = ref.current?.getElementsByClassName(
-			"a11y_alert",
-		)[0] as HTMLElement;
-		levelNav(event, ref, alertDiv, selectorType, multiSeries, nextSeries, series, selectedSeries);
-		let numberAux = keyHandler(
-			type,
-			event,
-			number,
-			ref,
-			selectorType,
-			insights,
-			insightsArray,
-			arrayConverted,
-			title,
-			descs,
-			series,
-			selectedSeries,
-			autoDescriptions,
-		);
-		setNumber(numberAux);
-	}
-
-	// features exclusive to bar charts (might be able to turn this more modular)
-	if (!selectorType) {
-		console.log("Type of chart not supported or no type given");
-	}
 	return (
-		<div
-			ref={ref}
-			onKeyDown={handleKeyDown}
-			className="a11y_chart"
-			data-testid="a11y_chart"
-			role="form"
-		>
-			<p
-				style={{ textIndent: "-10000px" }}
-				className="a11y_desc visually-hidden"
-				data-testid="a11y_desc"
-				onFocus={handleFocus}
-				onBlur={handleBlur}
+		<>
+			<div
+				ref={chartRef}
+				onKeyDown={(event) => {
+					handleKeyDown({
+						event,
+						alertDivRef,
+						type,
+						number,
+						chartRef,
+						elements,
+						selectedSeries,
+						series,
+						selectorType,
+						setSelectedSeries,
+						setNumber,
+						setDescriptionContent,
+						insights,
+						insightsArray,
+						arrayConverted,
+						title,
+						descs,
+						autoDescriptions,
+						multiSeries,
+					});
+				}}
+				className="a11y_chart"
+				data-testid="a11y_chart"
+				role="form"
```

Comment:
Wait, if you are assigning this to the role of form, why not using the actual form html element? What is stopping you from doing that?

If it is the submission event on the form you can always prevent that.
Otherwise it doesn't make much sense to me to be hijacking the semantics of a html div element with a new role when, in fact, we already have an element precisely for that.

Reply:
NONONO
The addition of `form` was a failed workaround of making the Focus mode on NVDA and JAWS activate automatically. This Focus mode is necessary for the users to navigate using the arrow keys inside the charts. Not working, i say we can just take this out.